src/voice-broadcast/components/molecules/VoiceBroadcastPreRecordingPip.tsx

```diff
@@ -28,34 +28,55 @@ interface Props {
     voiceBroadcastPreRecording: VoiceBroadcastPreRecording;
 }
 
+interface State {
+    showDeviceSelect: boolean;
+    disableStartButton: boolean;
+}
+
 export const VoiceBroadcastPreRecordingPip: React.FC<Props> = ({ voiceBroadcastPreRecording }) => {
     const pipRef = useRef<HTMLDivElement | null>(null);
     const { currentDevice, currentDeviceLabel, devices, setDevice } = useAudioDeviceSelection();
-    const [showDeviceSelect, setShowDeviceSelect] = useState<boolean>(false);
+    const [state, setState] = useState<State>({
+        showDeviceSelect: false,
+        disableStartButton: false,
+    });
 
-    const onDeviceSelect = (device: MediaDeviceInfo | null) => {
-        setShowDeviceSelect(false);
+    const onDeviceSelect = (device: MediaDeviceInfo) => {
+        setState({
+            ...state,
+            showDeviceSelect: false,
+        });
```

Comment:
```suggestion
        setState(state => ({
            ...state,
            showDeviceSelect: false,
        }));
```

This will be more robust in case React decides to skip a render